src/mrpro/operators/PCACompressionOp.py

```diff
@@ -0,0 +1,84 @@
+"""PCA Compression Operator."""
+
+import einops
+import torch
+from einops import repeat
+
+from mrpro.operators.LinearOperator import LinearOperator
+
+
+class PCACompressionOp(LinearOperator):
+    """PCA based compression operator."""
+
+    def __init__(
+        self,
+        data: torch.Tensor,
+        n_components: int,
+    ):
+        """Construct a PCA based compression operator.
+
+        The operator carries out an SVD followed by a threshold of the n_components largest values along the last
+        dimension of a data with shape (*other, joint_dim, compression_dim). A single SVD is carried out for everything
+        along joint_dim. Other are batch dimensions.
+
+        Consider combining this operator with RearrangeOp to make sure the data is in the correct shape before applying.
+
+        Parameters
+        ----------
+        data
+            Data to be used to find the principal components of shape (*other, joint_dim, compression_dim)
```

Comment:
Maybe change to:
`Data of shape (*other, joint_dim, compression_dim) to be used to find the principal components. `
Else sounds like the components have that shape...